packages/legend-studio-extension-query-builder/src/components/QueryBuilderDialog.tsx

```diff
@@ -42,13 +46,37 @@ export const QueryBuilderDialog = observer(() => {
   );
   const [isMaximized, setIsMaximized] = useState(false);
   const toggleMaximize = (): void => setIsMaximized(!isMaximized);
+  const USER_ATTESTATION_MESSAGE =
+    'Unsaved changes to your query will be lost if you continue. Are you sure?';
+
   const closeQueryBuilder = (): void => {
     flowResult(
       queryBuilderExtensionState.setEmbeddedQueryBuilderMode(undefined),
     ).catch(applicationStore.alertUnhandledError);
     queryBuilderExtensionState.reset();
   };
 
+  const confirmCloseQueryBuilder = (): void => {
+    applicationStore.setActionAlertInfo({
+      message: USER_ATTESTATION_MESSAGE,
+      type: ActionAlertType.CAUTION,
+      actions: [
+        {
+          label: 'Accept',
```

Comment:
Also, let's change the language here to sound less like compliance training. We're simply asking the users if they are aware and want to proceed. So `Proceed`, `Cancel`